external/blazegraph-service.js

```diff
@@ -0,0 +1,211 @@
+const axios = require('axios');
+const qs = require('qs');
+const constants = require('../modules/constants');
+
+class BlazegraphService {
+    constructor(config) {
+        this.config = config;
+    }
+
+    async initialize(logger) {
+        this.logger = logger;
+        this.logger.info('Blazegraph module initialized successfully');
+    }
+
+    async insert(triples, rootHash) {
+        const askQuery = `ASK WHERE { GRAPH <${rootHash}> { ?s ?p ?o } }`;
+        const exists = await this.ask(askQuery);
+        if (!exists) {
+            const config = {
+                method: 'post',
+                url: `http://localhost:9999/blazegraph/sparql?context-uri=${rootHash}`,
+                headers: {
+                    'Content-Type': 'text/x-nquads',
+                },
+                data: triples,
+            };
+
+            await axios(config).then((response) => true)
+                .catch((error) => {
+                    this.logger.error(`Failed to write into Blazegraph: ${error} - ${error.stack}`);
+                    return false;
+                });
+        }
+        // TODO: else -> Should log if we already have data
+    }
+
+    async execute(query) {
+        return new Promise(async (accept, reject) => {
+            const data = qs.stringify({
+                query,
+            });
+            const config = {
```

Comment:
Initialize config object in a constructor and use it through code so that we don't repeat a code. Use constants for url-s